src/sqlfluff/core/default_config.cfg

```diff
@@ -23,6 +23,8 @@ ignore = None
 # Warn only for rule codes (one of more rule codes, seperated by commas: e.g. LT01,LT02)
 # Also works for templating and parsing errors by using TMP or PRS
 warnings = None
+# Whether to warn about unneeded '-- noqa:' comments.
+warn_unused_ignores = False
```

Comment:
I wonder if this should be defaulted to True?

I know I had concerns, but as long as it's a warning and doesn't exit error I think that's OK.

Reply:
Good question 🤔 . I'll merge it default "off" for now - because it's new and that matches the behaviour of mypy most closely. It also has the benefit of allowing more extensive testing for now.